shell/platform/fuchsia/flutter/session_connection.h

```diff
@@ -82,7 +82,14 @@ class SessionConnection final {
   uint64_t next_present_session_trace_id_ = 0;
   uint64_t processed_present_session_trace_id_ = 0;
 
-  bool presentation_callback_pending_ = false;
+  // The maximum number of frames Flutter sent to Scenic that it can have
+  // outstanding at any time. This is equivalent to how many times it has
+  // called Present2() before receiving an OnFramePresented() event.
+  static constexpr int kMaxFramesInFlight = 2;
```

Comment:
Is there a reason that this is not higher/lower? My understanding is that the higher this number is, the more frames we will process while we wait for vsync events. Is this 2 because [Pipeline](https://github.com/flutter/engine/blob/master/shell/common/animator.cc#L40) depth is 2?

Reply:
It's 2 just for now to fit the current behavior, which uses a bool to essentially allow at most 2 frames in flight. My plan is to change this to be 3 in an upcoming CL so that we effectively have a third framebuffer.